src/org/rocketproplab/marginalstability/flightcomputer/commands/HeartbeatCommand.java

```diff
@@ -0,0 +1,124 @@
+package org.rocketproplab.marginalstability.flightcomputer.commands;
+
+import org.rocketproplab.marginalstability.flightcomputer.Settings;
+import org.rocketproplab.marginalstability.flightcomputer.Time;
+import org.rocketproplab.marginalstability.flightcomputer.subsystems.Subsystem;
+import org.rocketproplab.marginalstability.flightcomputer.subsystems.Telemetry;
+
+/**
+ * This is the Heartbeat Command class that implements the Command interface.
+ * The main capabilities of this class includes checking the start time and
+ * sending a periodic "heartbeat" signal every second.
+ * 
+ * @author Hemanth Battu
+ *
+ */
+public class HeartbeatCommand implements Command {
+  private static final Subsystem[] EMPTY_ARRAY = {};
+  private int                      HBcounter;
+  private double                   startTime;
+  private double                   currentTime;
+  private Time                     time;
+  private Telemetry                telemetry;
+
+  /**
+   * Creates a new HeartbeatCommand object using Time and Telemetry objects.
+   * 
+   * @param time      the Time object to use for checking time
+   * @param telemetry the Telemetry object used to send heartbeat
+   */
+  public HeartbeatCommand(Time time, Telemetry telemetry) {
+    this.time      = time;
+    this.telemetry = telemetry;
+  }
+
+  /**
+   * Getter method to return start time.
+   * 
+   * @return startTime
+   */
+  public double getStartTime() {
+    return startTime;
+  }
+
+  /**
+   * Setter method to set the start time.
+   * 
+   * @param startTime input to set start time with
+   */
+  public void setStartTime(double startTime) {
+    this.startTime = startTime;
+  }
+
+  /**
+   * Getter method to return current time.
+   * 
+   * @return currentTime
+   */
+  public double getcurrentTime() {
+    return currentTime;
+  }
+
+  /**
+   * Setter method to set the current time.
+   * 
+   * @param currentTime input to set the current time with
+   */
+  public void setCurrentTime(double currentTime) {

```

Comment:
it looks like set current time is in the same situation as set start time. I suggest making it private.
```suggestion
  private void setCurrentTime(double currentTime) {
```